### PR TITLE
lock whole block device file running makefs

### PR DESCRIPTION
**Pull request by poettering**
A fix for #10179.

### CODE REVIEW

File: src/partition/makefs.c [hunk 1]
```diff
@@ -54,28 +57,31 @@ static int run(int argc, char *argv[]) {
...
         if (stat(device, &st) < 0)
                 return log_error_errno(errno, "Failed to stat \"%s\": %m", device);
 
-        if (!S_ISBLK(st.st_mode))
+        if (S_ISBLK(st.st_mode)) {
+                /* Lock the device so that udev doesn't interfere with our work */
+
+                lock_fd = lock_whole_block_device(st.st_rdev, LOCK_EX);
+                if (lock_fd < 0)
+                        return log_error_errno(r, "Failed to lock whole block device of \"%s\": %m", device);
```
Comment (mrc0mmand) [inline src/partition/makefs.c]: I guess this should be `return log_error_errno(lock_fd, ...`

Comment (poettering) [reply src/partition/makefs.c]: indeed

Comment (poettering) [reply src/partition/makefs.c]: indeed!

**Comment by poettering** [comment]
new version force pushed with issue @mrc0mmand found fixed

File: src/cryptsetup/cryptsetup-generator.c [hunk 2]
```diff
@@ -367,12 +367,12 @@ static int create_disk(
 
         if (tmp)
                 fprintf(f,
-                        "ExecStartPost=/sbin/mke2fs '/dev/mapper/%s'\n",
+                        "ExecStartPost=" ROOTLIBEXECDIR "/systemd-makefs ext2 '/dev/mapper/%s'\n",
```
Comment (keszybz) [inline src/cryptsetup/cryptsetup-generator.c:370]: Interesting! We talked about this for some time, I never suspected the change would be so tiny ;)

**Comment by flokli** [comment]
> I was thinking whether to instead pull in a regular
systemd-makefs@.service instance, but I couldn't come up with a reason
to, and thus opted for just doing the minimal patch and just replacing
the simply mkfs calls.

`systemd-makefs@.service` does support being configured after systemd is built, by the regular drop-ins (configuring `PATH` that `systemd-makefs` is executed with).

Also, didn't this PR handle

> * cryptsetup: reimplement the mkswap/mke2fs in cryptsetup-generator to use 
  systemd-makefs.service instead.

from `TODO`?

**Pull request merged**